pkg/query-service/app/metrics/v4/delta/timeseries.go

```diff
@@ -118,3 +160,37 @@ func prepareMetricQueryDeltaTimeSeries(start, end, step int64, mq *v3.BuilderQue
 
 	return query, nil
 }
+
+// canShortCircuit returns true if we can use the optimized query
+// for the given query
+// This is used to avoid the group by fingerprint thus improving the performance
+// for certain queries
+// cases where we can short circuit:
+// 1. time aggregation = (rate|increase) and space aggregation = sum
+//   - rate = sum(value)/step, increase = sum(value) - sum of sums is same as sum of all values
+//
+// 2. time aggregation = sum and space aggregation = sum
+//   - sum of sums is same as sum of all values
+//
+// 3. time aggregation = min and space aggregation = min
+//   - min of mins is same as min of all values
+//
+// 4. time aggregation = max and space aggregation = max
+//   - max of maxs is same as max of all values
+//
+// all of this is true only for delta metrics
```

Comment:
Have we considered using cumulativeToDelta processor? If there is a perf gain, we should

Reply:
We can't use it in our collector instance because it is a stateful processor and the same data point has to arrive at the same collector for it to work correctly which is not guaranteed so we are not going to use it.

I have mentioned this elsewhere in the delta EPIC issue, when we roll out the delta and make it the default we will update our docs to send cumulativetodelta on client collector instances and SDK instructions via [ENV](https://github.com/open-telemetry/opentelemetry-specification/blob/12cd1b1a4750a287282c4bfcd713de38d6716700/specification/metrics/sdk_exporters/otlp.md#additional-configuration) to prefer delta temporality.

Reply:
ok..so we are good to receive delta but not convert cumulative to delta